context/AuthProvider.tsx

```diff
@@ -33,39 +34,39 @@ export const AuthProvider = ({ children }: PropsWithChildren) => {
   const [user, setUser] = useState<User | null>();
   const [session, setSession] = useState<Session | null>(null);
   const [initialized, setInitialized] = useState<boolean>(false);
-  const [userStatus, setUserStatus] = useState<UserStatus>();
-  const { phoneNumber, setUserInfo } = useUserStore();
+  const [userStatus, setUserStatus] = useState<UserStatus>("New");
+  const { setUserInfo, userInfo } = useUserStore();
 
   useEffect(() => {
+    const loadUserStatus = async () => {
+      const storedStatus = await AsyncStorage.getItem("userStatus");
+      if (storedStatus) {
+        setUserStatus(storedStatus as UserStatus);
+      }
+    };
+
+    loadUserStatus();
+
     // Listen for changes to authentication state
     const { data } = supabase.auth.onAuthStateChange(async (_, session) => {
       setSession(session);
       setUser(session ? session.user : null);
       setInitialized(true);
+
+      if (session && session.user) {
+        const userData = await getUserByPhoneNumber(`+${session.user.phone!}`);
```

Comment:
Might be worth using a lib instead of adding the + to get e164 format, e.g. https://www.npmjs.com/package/awesome-phonenumber